server/channels/app/bot.go

```diff
@@ -38,29 +38,32 @@ func (a *App) EnsureBot(rctx request.CTX, pluginID string, bot *model.Bot) (stri
 		return "", err
 	}
 
-	// If the bot has already been created, use it
+	// If the bot has already been created, check whether it still exists and use it
 	if botIDBytes != nil {
 		botID := string(botIDBytes)
+		if _, err := a.GetBot(rctx, botID, true); err != nil {
```

Comment:
Golangci is complaining about this part
```
Error: channels/app/bot.go:44:9: shadow: declaration of "err" shadows declaration at line 36 (govet)
		if _, err := a.GetBot(rctx, botID, true); err != nil {
		      ^
make: *** [Makefile:290: golangci-lint] Error 1
```

I suggest that you rename the `*model.AppError` in L36 to `appErr` per https://github.com/mattermost/mattermost/pull/26545#discussion_r1542478728.